models/pull.go

```diff
@@ -1128,10 +1158,29 @@ func AddTestPullRequestTask(doer *User, repoID int64, branch string, isSync bool
 	}
 
 	if isSync {
-		if err = PullRequestList(prs).LoadAttributes(); err != nil {
+		requests := PullRequestList(prs)
+		if err = requests.LoadAttributes(); err != nil {
 			log.Error(4, "PullRequestList.LoadAttributes: %v", err)
 		}
+		var gitRepo *git.Repository
+		repo, err := GetRepositoryByID(repoID)
+		if err != nil {
+			log.Error(4, "GetRepositoryByID: %v", err)
+			goto REQUIRED_PROCEDURE
+		}
+		gitRepo, err = git.OpenRepository(repo.RepoPath())
+		if err != nil {
+			log.Error(4, "git.OpenRepository: %v", err)
+			goto REQUIRED_PROCEDURE
```

Comment:
not sure how much `goto` is used in this codebase but couldn't you put the next goroutine in the `} else {` of this `if err != nil {}` ?